pkg/datapath/linux/probes/probes.go

```diff
@@ -422,3 +395,101 @@ func (p *ProbeManager) KernelConfigAvailable() bool {
 
 	return true
 }
+
+// HaveMapType is a wrapper around features.HaveMapType() to check if a certain
+// BPF map type is supported by the kernel.
+// On unexpected probe results this function will terminate with log.Fatal().
+func HaveMapType(mt ebpf.MapType) error {
+	err := features.HaveMapType(mt)
+	if errors.Is(err, ebpf.ErrNotSupported) {
+		return err
+	}
+	if err != nil {
+		log.WithError(err).Fatalf("failed to probe MapType: %s", mt.String())
+	}
+	return nil
+}
+
+// HaveProgramType is a wrapper around features.HaveProgramType() to check
+// if a certain BPF program type is supported by the kernel.
+// On unexpected probe results this function will terminate with log.Fatal().
+func HaveProgramType(pt ebpf.ProgramType) error {
+	err := features.HaveProgramType(pt)
+	if errors.Is(err, ebpf.ErrNotSupported) {
+		return err
+	}
+	if err != nil {
+		log.WithError(err).Fatalf("failed to probe ProgramType: %s", pt.String())
```

Comment:
Again, we can't just fatal out here. SkMsg is only supported in 4.19+ IIRC, but we will support older kernel versions for a while yet. This should return another declared error type.

Reply:
Ditto.